src/test/java/io/pravega/connectors/flink/PravegaInputSplitTest.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.connectors.flink;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+import io.pravega.client.batch.SegmentRange;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public class PravegaInputSplitTest {
+
+    @Mock
+    SegmentRange  range1;
+
+    @Mock
+    SegmentRange range2;
+
+    @Mock
+    SegmentRange range3;
+
+    @Mock
+    PravegaInputSplit mockSplitNullSegmentRange;
+
+    @Mock
+    PravegaInputSplit mockSplitNullSegmentRange2;
+
+
+    @Before
+    public void setupMocks() throws Exception {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void testEquals() {
+        PravegaInputSplit split1 = new PravegaInputSplit(12345, range1);
+        PravegaInputSplit split2 = new PravegaInputSplit(12345, range1);
+
+        // compare one to itself
+        assertTrue(split1.equals(split1));
+
+        // compare one to the other
+        assertTrue(split1.equals(split2));
+
+    }
+
+    @Test
+    public void testNotEquals() {
```

Comment:
can you also make sure to do null checks validation? 
range1 == range2 == null
range1 == null && range1 != null
range1 != null && range2 == null

Reply:
I don't think range1 or range2 can be null. The constructor for split input has the following precondition check which makes it impossible to create one with a null range in test:
Preconditions.checkNotNull(segmentRange, "segmentRange");
SegmentRange is an interface with SegmentRangleImpl having the following annotations:
@EqualsAndHashCode
So I believe those are covered by cases that check segmentRange.equals(segmentRange). since both null should be true and one null and one not should yield false. 
I tried doing these cases with Mokito with mock ranges and when (...).thenReturn(null), which worked for 2 of them but not the third. Its been awhile but I think it was the middle one where it couldn't appropriately resolve one of the mocks. If you can come up with a scheme that will work, I'd be happy to implement it.